frontend/src/components/bucket/BucketPermissionUser.vue

```diff
@@ -0,0 +1,109 @@
+<script setup lang="ts">
+import { onMounted, ref } from 'vue';
+import { storeToRefs } from 'pinia';
+import Button from 'primevue/button';
+import Dropdown from 'primevue/dropdown';
+import { useBucketStore, useUserStore } from '@/store';
+import type { User } from '@/interfaces';
+
+const userStore = useUserStore();
+
+const { permissions } = storeToRefs(useBucketStore());
+const { userSearch } = storeToRefs(useUserStore());
+
+onMounted(() => {
+  reset();
+});
+
+const isAddDisabled: any = ref(true);
+const showUserExistError: any = ref(false);
+const selectedUser = ref();
+
+const emit = defineEmits(['cancel-bucket-permission-user']);
+
+const cancelBucketPermissionUser = async () => {
+  isAddDisabled.value = true;
+  reset();
+  emit('cancel-bucket-permission-user');
+};
+
+const ifUserExist = () => {
+  return permissions.value.some(e => e.fullName === selectedUser.value.fullName);
+};
+
+const onInput = async (event: any) => {
+  showUserExistError.value = ifUserExist();
+  var inputValue = event.target.value;
+  if(inputValue.length > 0) {
+    await userStore.searchUsers({ search: inputValue });
```

Comment:
i cant remember if we are searching by just email address (I think so) in which case maybe this could be something like `userStore.searchUsers({ email: inputValue })` (?) i guess i depends on the searchUser() function.

Reply:
it might be something to tackle in another PR but i think we talked about filtering searchUsers to `&idp=idir` unless inputValue is a full email address